src/pages/_app.js

```diff
@@ -1,29 +1,45 @@
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import { useRouter } from "next/router";
 import transactionsData from "@/src/data/transactions.json";
 import { interFont } from "@/src/styles/font";
 import "@/src/styles/global.css";
 import { uid } from "uid";
 
 export default function App({ Component, pageProps }) {
-  const [initialData, setInitialData] = useState(transactionsData);
+  const [initialData, setInitialData] = useState([]);
   const [mode, setMode] = useState("default");
   const [action, setAction] = useState("default");
   const router = useRouter();
 
-  function handleMode(newMode) {
-    setMode(newMode);
+  useEffect(() => {
+    const localData = localStorage.getItem("transactions");
+    if (localData) {
+      setInitialData(JSON.parse(localData));
+    } else {
+      localStorage.setItem("transactions", JSON.stringify(transactions));
+      setInitialData(transactions);
+    }
+  }, []);
```

Comment:
instead of useEffect() if and else this simple code should do 

 import useLocalStorageState from "use-local-storage-state";

const [transactions, setTransactions] = useLocalStorageState("transactions", {
    defaultValue: {initialData},
  });

Reply:
that is absolutely right! thank you very much